test/OpenIddict.Server.Tests/OpenIddictServerBuilderTests.cs

```diff
@@ -0,0 +1,2000 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using Moq;
+using OpenIddict.Abstractions;
+using Xunit;
+using static OpenIddict.Server.OpenIddictServerEvents;
+
+namespace OpenIddict.Server.Tests
+{
+    public class OpenIddictServerBuilderTests
+    {
+        [Fact]
+        public void Constructor_ThrowsAnExceptionForNullServices()
+        {
+            // Arrange
+            var services = (IServiceCollection)null;
+
+            // Act and assert
+            var exception = Assert.Throws<ArgumentNullException>(() => new OpenIddictServerBuilder(services));
+
+            Assert.Equal("services", exception.ParamName);
+        }
+
+        private class CustomContext : BaseContext
+        {
+            /// <summary>
+            /// Creates a new instance of the <see cref="BaseContext"/> class.
+            /// </summary>
+            public CustomContext(OpenIddictServerTransaction transaction) : base(transaction) { }
+        }
+
+        private class CustomHandler : IOpenIddictServerHandler<CustomContext>
+        {
+            /// <summary>
+            /// Processes the event.
+            /// </summary>
+            /// <param name="context">The context associated with the event to process.</param>
+            /// <returns>
+            /// A <see cref="ValueTask"/> that can be used to monitor the asynchronous operation.
+            /// </returns>
+            public ValueTask HandleAsync(CustomContext context)
+            {
+                return new ValueTask();
+            }
+        }
+
+        [Fact]
+        public void AddEventHandler_ThrowsAnExceptionWhenConfigurationIsNull()
+        {
+            // Arrange
+            var services = CreateServices();
+            var builder = CreateBuilder(services);
+            Action<OpenIddictServerHandlerDescriptor.Builder<CustomContext>> configuration = null;
+
+            // Act
+            Action action = () => builder.AddEventHandler<CustomContext>(configuration);
```

Comment:
Technically, the exception is not thrown at this stage, so it's weird to put it in the "act" phase. Could you please combine the act and assert into a single "Act and assert" phase?

Reply:
Updated all cases where this was being done in this way.